ndk-patches/27/stdio.h.patch

```diff
@@ -1,10 +1,12 @@
 --- ./usr/include/stdio.h.orig	2021-08-16 11:48:44.308954400 +0200
 +++ ./usr/include/stdio.h	2021-08-16 11:50:00.438924014 +0200
-@@ -44,6 +44,9 @@
+@@ -44,6 +44,11 @@
  #include <stdarg.h>
  #include <stddef.h>
  
++#if !defined(__swift__)
```

Comment:
Needed this last commit for the upcoming Swift 6, shouldn't affect anything else.

Reply:
Could you explain the reason for this change or what issues are fixed?

Reply:
Sure, wasn't sure if anyone would be interested. 😄

Swift 6 adds [a fully modularized `android.modulemap`](https://github.com/swiftlang/swift/pull/72161/files#diff-1fd3199a3e1289fac4557405173ac51a19939b34f76f89676fa38833761398bd), as opposed to [the `glibc.modulemap`](https://github.com/swiftlang/swift/blob/main/stdlib/public/Platform/glibc.modulemap.gyb) we reuse from linux in the current Swift 5.10 package. This change is primarily to help with [the new C++ interoperability features of Swift](https://www.swift.org/documentation/cxx-interop/), which require a modularized libc module map to go with [the libc++ module.modulemap from the NDK that we currently ship](https://android.googlesource.com/platform/prebuilts/clang/host/linux-x86/+/refs/heads/main/clang-r522817/include/c++/v1/module.modulemap).

The problem is these additional headers we patch in for Termux break that modularity, in ways I don't understand. So, I found that [the Swift compiler sets `__swift__` when importing these headers](https://github.com/swiftlang/swift/blob/main/lib/ClangImporter/ClangImporter.cpp#L510) and undefined these when building for Swift, including the `ctermid()` inline function definition that needs that header.